packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -187,6 +199,52 @@ class InputDateTimePicker extends React.Component {
 		this.updateDateTime(date, time, fullString, errMsg);
 	}
 
+	onFocusInput() {
+		this.switchDropdownVisibility(true);
+	}
+
+	setContainerRef(ref) {
+		this.containerRef = ref;
+	}
+
+	setDropdownWrapperRef(ref) {
+		this.dropdownWrapperRef = ref;
+	}
+
+	mountDocumentHandler() {
+		document.addEventListener('click', this.documentHandler);
+		document.addEventListener('focus', this.documentHandler);
+	}
+
+	unmountDocumentHandler() {
+		document.removeEventListener('click', this.documentHandler);
+		document.removeEventListener('focus', this.documentHandler);
+	}
+
+	documentHandler(e) {
+		const isActionOutOfComponent = e.path.every(node => node !== this.containerRef);
+
+		if (isActionOutOfComponent) {
+			this.switchDropdownVisibility(false);
+		}
+	}
+
+	switchDropdownVisibility(isShown) {
+		if (this.state.isDropdownShown === isShown) {
+			return;
+		}
+
+		if (isShown) {
```

Comment:
You should set this as `setState` callback

Reply:
done

Reply:
Good catch